torch/_inductor/lowering.py

```diff
@@ -1074,8 +1074,8 @@ def inner_fn(idx):
         zero_point = zero_points_loader(channel_idx)
         qmin, qmax = _create_constants(quant_min, quant_max, dtype=torch.float32)
 
-        inv_scale = ops.reciprocal(scale)
-        val = ops.round(input * inv_scale) + zero_point
+        inv_scale = ops.to_dtype(ops.reciprocal(scale), torch.float32)
```

Comment:
Can we check the dtype of `scale`/`zp` and only do convert when necessary? Same as below.

Reply:
Hi @leslie-fang-intel. Looks like even if I don't convert scale to float explicitly by `ops.to_dtype`, it is done in the generated code:
```cpp
auto tmp1 = in_ptr1[static_cast<long>(x0)]; // const double* in_ptr1
auto tmp23 = c10::convert<float>(tmp1); // convert scale to float. This line appears even if I don't write `to_dtype` in the lowering path
auto tmp24 = at::vec::Vectorized<float>(tmp23);
auto tmp25 = tmp22 * tmp24;
```
Do you know why?
And for zero point, I guess it is not a problem to convert because we can assume zero point is not float in most cases.

Reply:
https://github.com/pytorch/pytorch/blob/cead0363a87cbea501c27a637d06580f75ef12f7/torch/_inductor/codegen/cpp.py#L1018 Codegen wrapper will align the scalar to the vec dtype.

Reply:
Ok. I have discussed with Jiong. We will do dtype conversion explicitly.